### PR TITLE
fix(docs): miscellaneous docs reorg

### PR DESCRIPTION
**Pull request by amberleyromo**
##  Description

- Link accessibility guide in the "building themes" guide
- Move theme API doc to api-reference section, create redirect
- Add theme tutorials section to tutorial index
- Rearrange docs landing page to highlight two main pathways.

### CODE REVIEW

File: www/gatsby-node.js [hunk 1]
```diff
@@ -46,6 +46,12 @@ const slugToAnchor = slug =>
 exports.createPages = ({ graphql, actions, reporter }) => {
   const { createPage, createRedirect } = actions
 
+  createRedirect({
+    fromPath: `/docs/themes/api-reference`, // Merged Component CSS and CSS Modules
+    toPath: `/docs/themes-api/`,
```
Comment (johno) [inline www/gatsby-node.js]: It looks like the path down below is `/theme-api/` rather than `/themes-api/`, not sure if that's intentional?

Comment (amberleyromo) [reply www/gatsby-node.js]: oh, nope oversight. gracias.

Comment (marcysutton) [reply www/gatsby-node.js]: That's the only thing that needs to be updated, IMO!

Comment (amberleyromo) [reply www/gatsby-node.js]: fixed!

**Comment by marcysutton** [comment]
One thing we should add is a link to the Theme API docs in their new location. I think the Building Themes page would be a good place for that, what do you think? https://www.gatsbyjs.org/docs/themes/building-themes

**Comment by marcysutton** [comment]
Nice work on this, a great collection of changes!

**Pull request merged**